src/compiler/core.ts

```diff
@@ -459,6 +459,14 @@ module ts {
             if (path.charCodeAt(2) === CharacterCodes.slash) return 3;
             return 2;
         }
+        // Per RFC 1738'file' URI schema has a shape file://<host>/<path>
+        // if <host> is omitted then it is assumed that host value is'localhost',
+        // however slash after the omitted <host> is not removed.
+        // file:///folder1/file1 - this is correct URI
+        // file://folder2/file2 - this is incorrect URI
+        if (path.lastIndexOf("file:///", 0) === 0) {

```

Comment:
Why `lastIndexOf`? Does `file://file://foo` need to be special-cased in some way?